src/data/nat/fib.lean

```diff
@@ -148,4 +153,10 @@ end
 lemma fib_dvd (m n : ℕ) (h : m ∣ n) : fib m ∣ fib n :=
 by rwa [gcd_eq_left_iff_dvd, ← fib_gcd, gcd_eq_left_iff_dvd.mp]
 
+lemma fib_succ_eq_sum_choose :
+  ∀ (n : ℕ), fib (n + 1) = ∑ p in finset.nat.antidiagonal n, choose p.1 p.2 :=
+two_step_induction rfl rfl (λ n h1 h2, by
+{ rw [fib_add_two, h1, h2, finset.nat.antidiagonal_succ_succ', finset.nat.antidiagonal_succ'],
+  simp [choose_succ_succ, finset.sum_add_distrib, add_left_comm] })
```

Comment:
I wonder if this proof would be easier to follow if `antidiagonal_succ` etc used `cons` instead of `insert` - that way you'd get a proof that the element is disjoint from the rest of the antidiagonal at the same time.

Reply:
I think that `antidiagonal` was added before `finset.cons`.